### PR TITLE
fix: add missing clientCertificate type to ResolvedConfigOptions

### LINKED ISSUE #17799: Missing types for clientCertificate
### Current behavior

There is a `clientCertificates` option. https://docs.cypress.io/guides/references/client-certificates#Syntax

It should include in the type definitions: https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts#L2491

### Desired behavior

Included in types.

### Test code to reproduce

```ts
const C: Cypress.ResolvedConfigOptions = {
  clientCertificates: // <- no completion or hint
}
```

### Cypress Version

8.0.0

### Other

This is a good first issue for a new contributor or a new employee. Although adding the type is simple enough, it's a good way to learn how to set up Cypress locally, make a PR, get a review, etc. Let's encourage a new contributor to take this one.

### PR DESCRIPTION
**Pull request by ImCesar**
- Closes https://github.com/cypress-io/cypress/issues/17799

### User facing changelog
Add missing clientCertificate field type definition to ResolvedConfigOptions

### Additional details
Provides code completion for clientCertificate

### PR Tasks
- [x] Have tests been added/updated?
- [ ] Has the original issue or this PR been tagged with a release in ZenHub? 
- [x] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [x] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: cli/types/cypress.d.ts [hunk 1]
```diff
@@ -2750,6 +2791,11 @@ declare namespace Cypress {
      * @default {}
      */
     e2e: Omit<ResolvedConfigOptions, TestingType>
+
+    /**
+     * An array of objects defining the certificates
+     */
+    clientCertificate: ClientCertificate[]
```
Comment (lmiller1990) [inline cli/types/cypress.d.ts:2798]:
```suggestion
    clientCertificates: ClientCertificate[]
```

**Review by lmiller1990 — CHANGES_REQUESTED** [review]
Just one change, made a recommendation.

I wonder if we could also use this new type in place of this one to avoid duplication? https://github.com/cypress-io/cypress/blob/956e1c56d2857519b0b6842ea4e35b938cb5ad85/packages/network/lib/client-certificates.ts#L99

**Review by lmiller1990 — APPROVED** [review]

**Comment by ImCesar** [comment]
> Just one change, made a recommendation.
> 
> I wonder if we could also use this new type in place of this one to avoid duplication?
> 
> https://github.com/cypress-io/cypress/blob/956e1c56d2857519b0b6842ea4e35b938cb5ad85/packages/network/lib/client-certificates.ts#L99

This one looks different than the type defined in the documentation. It seems to me that the one above defines the actual ClientCertificate as it's used in code and the one defined for this issue is how a ClientCertificate is defined for configuration.

**Comment by JessicaSachs** [comment]
LGTM, failing test is known flake.

**Pull request merged**